compute/compute-node.Dockerfile

```diff
@@ -1123,13 +1409,17 @@ RUN wget https://github.com/pgpartman/pg_partman/archive/refs/tags/v5.1.0.tar.gz
 # compile pg_mooncake extension
 #
 #########################################################################################
-FROM rust-extensions-build AS pg-mooncake-build
+FROM build-deps AS pg_mooncake-src
 ARG PG_VERSION
-
+WORKDIR /ext-src
```

Comment:
Could we deduplicate this line by putting into the build-deps layer?

Reply:
Hmm. That would be nice, but the `build-deps` is used as the base for other layers too, not just for downloading sources. So I wouldn't put it into `build-deps`. Perhaps we should introduce a new `download-src` layer, which contains just `wget` and other tools needed for the downloads, and that could set WORKDIR.

I feel it's probably better to be explicit with WORKDIR though, even if it's a bit more verbose. The [Dockerfile manual](https://docs.docker.com/reference/dockerfile/#workdir) also mentions that as a best practice.